### DISCUSSION TITLE
Question on PKCS7 Padding

### DISCUSSION THREAD

**Discussion by matthiasgeihs** [krzyzanowskim/CryptoSwift#1020] [Q&A]
https://github.com/krzyzanowskim/CryptoSwift/blob/db51c407d3be4a051484a141bf0bff36c43d3b1e/Sources/CryptoSwift/PKCS/PKCS7Padding.swift#L30-L35

PKCS7 padding has an if branch that seems unreachable, if I see it correctly.
In the above example, `padding = blockSize - (bytes.count % blockSize)`.
For `blockSize > 0`, `blockSize` is strictly greater `a % blockSize`, for any integer `a`.
Hence, `padding` can never be zero.

==> The branch `if padding == 0` should be removed.

Right?

(Might relate to https://github.com/krzyzanowskim/CryptoSwift/discussions/953.)

**Comment by krzyzanowskim** [comment]
it's 0 for bytes.count == blockSize, isn't it

**Reply by matthiasgeihs** [reply]
Assume `blockSize > 0`.

If `bytes.count == blockSize`, then `bytes.count % blockSize == 0`.
Hence, `padding = blockSize - (bytes.count % blockSize) = blockSize`.
Since `blockSize > 0`, `padding != 0`.
